dotCMS/src/main/java/com/dotmarketing/portlets/folders/business/FolderAPIImpl.java

```diff
@@ -50,613 +50,620 @@
+            throw new DotSecurityException("User " + user.getUserId() != null?user.getUserId():"" + " does not have permission to add to Host " + newParentHost.getHostname());
+        }
+
+        folderFactory.copy(folderToCopy, newParentHost);
+
+        this.systemEventsAPI.pushAsync(SystemEventType.COPY_FOLDER, new Payload(folderToCopy, Visibility.EXCLUDE_OWNER,
+                new ExcludeOwnerVerifierBean(user.getUserId(), PermissionAPI.PERMISSION_READ, Visibility.PERMISSION)));
+    }
+
+    @CloseDBIfOpened
+    public boolean exists(String folderInode) throws DotDataException {
+        return folderFactory.exists(folderInode);
+    }
+
+
+    @SuppressWarnings("unchecked")
+    @CloseDBIfOpened
+    public List<Inode> findMenuItems(Folder folder, User user, boolean respectFrontEndPermissions) throws DotStateException, DotDataException {
+        return folderFactory.getMenuItems(folder);
+    }
+
+    /**
+     * Takes a folder and a user and deletes all underneith assets User needs
+     * edit permssions on folder to delete everything undernieth
+     *
+     * @param folder
+     * @param user
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    @WrapInTransaction
+    public void delete(Folder folder, User user, boolean respectFrontEndPermissions) throws DotDataException, DotSecurityException {
+
+        String path = StringUtils.EMPTY;
+
+        if(folder==null || !UtilMethods.isSet(folder.getInode()) ){
+            Logger.debug(getClass(), "Cannot delete null folder");
+            return;
+        } else {
+            AdminLogger.log(this.getClass(), "delete", "Deleting folder with name " + (UtilMethods.isSet(folder.getName()) ? folder.getName() + " ": "name not set "), user);
+        }
+        if (!permissionAPI.doesUserHavePermission(folder, PermissionAPI.PERMISSION_EDIT, user, respectFrontEndPermissions)) {
+
+            Logger.debug(getClass(), "The user: " + user.getEmailAddress() + " does not have permission for: " + folder );
+            throw new DotSecurityException("User " + user.getUserId() != null?user.getUserId():"" + " does not have permission to edit Folder " + folder.getPath());
+        }
+
+
+        if(folder != null && FolderAPI.SYSTEM_FOLDER.equals(folder.getInode())) {
+
+            Logger.debug(getClass(), "Cannot delete null folder: " + folder.getInode());
+            throw new DotSecurityException("YOU CANNOT DELETE THE SYSTEM FOLDER");
+        }
+
+
+        // start transactional delete
+        try {
+
+            PermissionAPI papi = getPermissionAPI();
+            if (!papi.doesUserHavePermission(folder, PermissionAPI.PERMISSION_EDIT_PERMISSIONS, user)) {
+
+                Logger.error(this.getClass(), "User " + user.getUserId() != null?user.getUserId():"" + " does not have permission to Folder " + folder.getPath());
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Split this 162 characters long line (which is greater than 150 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00103)